run_analysis.sh

```diff
@@ -2,36 +2,67 @@
 
 ECOSYSTEM="$1"
 PACKAGE="$2"
+PKG_PATH="$3"
 
-if [[ $# != 2 ]]; then
-	echo "Usage: $0 (npm|packagist|pypi|rubygems) <package name>"
-	exit -1
+if [[ $# != 2 && $# != 3 ]]; then
+	echo "Usage: $0 (npm|packagist|pypi|rubygems) <package name> [/path/to/local/package.file]"
+	exit 255
 fi
 
 RESULTS_DIR="/tmp/results"
 LOGS_DIR="/tmp/dockertmp"
 
+LINE="-----------------------------------------"
+
 mkdir -p "$RESULTS_DIR"
 mkdir -p "$LOGS_DIR"
 
-docker run --cgroupns=host --privileged -ti \
-    -v "$RESULTS_DIR":/results \
-    -v "$LOGS_DIR":/tmp \
-    -v /var/lib/containers:/var/lib/containers \
-    gcr.io/ossf-malware-analysis/analysis analyze \
-    -package "$PACKAGE" -ecosystem "$ECOSYSTEM" \
-    -upload file:///results/
 
-EXIT_CODE=$?
+DOCKER_OPTS=(run --cgroupns=host --privileged -ti)
+
+DOCKER_MOUNTS=(-v /var/lib/containers:/var/lib/containers -v "$RESULTS_DIR":/results -v "$LOGS_DIR":/tmp)
+
+ANALYSIS_IMAGE=gcr.io/ossf-malware-analysis/analysis
+
+ANALYSIS_ARGS=(analyze -package "$PACKAGE" -ecosystem "$ECOSYSTEM" -upload file:///results/)
 
+echo $LINE
+
+if [[ -n "$PKG_PATH" ]]; then 
+	# local mode
+	PKG_FILE=$(basename "$PKG_PATH")
+	echo "Analysing $ECOSYSTEM package $PACKAGE from local file $PKG_FILE"
+	echo "Path: $PKG_PATH"
+
+	# mount local package file in root of docker image
+	DOCKER_MOUNTS+=(-v "$PKG_PATH:/$PKG_FILE")
+
+	# tell analyis to use mounted package file
+	ANALYSIS_ARGS+=(-local "/$PKG_FILE")
+else
+	# remote mode
+	echo "Analysing $ECOSYSTEM package $PACKAGE (remote)"
+fi
+
+echo $LINE
 echo
-if [[ $EXIT_CODE == 0 ]]; then
+
+sleep 1
```

Comment:
Actually, rather than sleeping, maybe it's worth printing a message at the end as well.

e.g
echo "Completed: analysis X package y, results in /X"

Reply:
Sure. There is currently a message printed at the end with the results path but I can repeat the package name there too